utils/dynamodb/formatResponse.js

```diff
@@ -0,0 +1,140 @@
+export const getAttributeMap = (data) => {
+  return data.Items
+  .filter(item => item.type.S === DYNAMO_DB_TYPES.attribute)
+  .reduce((result, item) => {
+    const attributeId = item.SK.S;
+    const attributeName = item.attributes.M.name.S;
+    const attributeType = item.attributes.M.type.S;
+    const attributeOrder = item.attributes.M.order.N;
+
+    result[attributeId] = {
+      name: attributeName,
+      type: attributeType,
+      order: parseInt(attributeOrder),
+    };
+
+    return result;
+  }, {});
+}
+
+// Formats the content from the DynamoDB response
+// Items here are those with `type = content` attribute
+export const formatContentData = (data, attributeMap) => {
+  const items = data.Items;
+
+  const content = items
+  .reduce((result, item) => {
+    const itemKeyParts = item.SK.S.split(DYNAMO_DB_ID_SEPARATOR);
+    const itemKey = itemKeyParts[1]; // get only the ulid
+    result[itemKey] = {
+      id: itemKey,
+      slug: item.slug.S,
+      status: item.status.S,
+    };
+
+    // checks if the content item has attributes
+    if (item.attributes) {
+      const formattedAttributes = Object.keys(item.attributes.M).reduce((formattedAttrs, attrKey) => {
+        const attributeName = attributeMap[attrKey]?.name;
+
+        if (attributeName) {
+          const attributeValue = item.attributes.M[attrKey].S;
+          formattedAttrs[attributeName] = {
+            value: attributeValue,
+            type: attributeMap[attrKey].type,
+            order: attributeMap[attrKey].order
+          };
+        }
+        
+        return formattedAttrs;
+      }, {});
+
+      const sortedAttributes = Object
+        .keys(formattedAttributes)
+        .sort((a, b) => formattedAttributes[a].order - formattedAttributes[b].order)
+        .reduce((sorted, attributeName) => {
+          sorted[attributeName] = formatAttributeValue(formattedAttributes[attributeName]); 
+          return sorted;
+        }, {});
+
+      result[itemKey] = {
+        ...result[itemKey],
+        ...sortedAttributes,
+      };
+    }
+
+    return result;
+  }, {});
+
+  return content;
+}
+
+// Formats the attributes 
+export const formatAttributesData = (data) => {
+  const items = data.Items;
+  const attributes = items
+    .filter(item => item.type.S === DYNAMO_DB_TYPES.attribute)
+    .reduce((result, item) => {
+      const attributeName = item.attributes.M.name.S;
+      const attributeType = item.attributes.M.type.S;
+      const attributeOrder = item.attributes.M.order.N;
+      
+      result[attributeName] = {
+        type: attributeType,
+        order: parseInt(attributeOrder),
+      };
+      
+      return result;
+    }, {});
+  
+  // Sorts by order 
+  const sortedAttributes = Object
+    .keys(attributes)
+    .sort((a, b) => attributes[a].order - attributes[b].order)
+    .reduce((sortedResult, attributeName) => {
+      sortedResult[attributeName] = attributes[attributeName];
+      return sortedResult;
+    }, {});
```

Comment:
We should avoid sorting in Next.js. We need to maximize DynamoDB's sort keys and global sort index to push the sorting operation in the DynamoDB side. 

This will get slower and slower as more entries are added to the database.